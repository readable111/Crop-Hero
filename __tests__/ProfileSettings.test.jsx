/****
 * @author Daniel Moreno
 * @reviewer
 * @tester 
 ***/

import React from "react";
import {
    render,
    waitFor,
    screen,
    fireEvent,
} from "@testing-library/react-native";
import SettingsProfile from "../app/profile-settings"
import Colors from "../assets/Color";
import AsyncStorage from "@react-native-async-storage/async-storage";
import renderer from "react-test-renderer";
import * as Font from 'expo-font';
jest.useFakeTimers();

jest.mock('expo-font', ()=>({
    useFonts: jest.fn(),
    loadAsync: jest.fn().mockResolvedValue(true),
    isLoaded: jest.fn(()=>true)
}))



describe('Light Mode <SettingsProfile/>', () =>{
    beforeEach(async () => {
        await AsyncStorage.setItem("dark_mode_setting", "false");
        Font.useFonts.mockReturnValue([true, false])
        const [fontsLoaded, fontError] = Font.useFonts()
        expect(fontsLoaded).toBe(true)
        expect(fontError).toBe(false)
    });

    test('renders correctly', () =>{
        const tree = render(<SettingsProfile/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('Dark Mode <SettingsProfile/>', () =>{
    beforeEach(async () => {
        await AsyncStorage.setItem("dark_mode_setting", "true");
        Font.useFonts.mockReturnValue([true, false])
        const [fontsLoaded, fontError] = Font.useFonts()
        expect(fontsLoaded).toBe(true)
        expect(fontError).toBe(false)
    });

    test('renders correctly', () =>{
        const tree = render(<SettingsProfile/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('No Async <SettingsProfile/>', () =>{
    beforeEach(async () => {
        await AsyncStorage.setItem("dark_mode_setting", "");
        Font.useFonts.mockReturnValue([true, false])
        const [fontsLoaded, fontError] = Font.useFonts()
        expect(fontsLoaded).toBe(true)
        expect(fontError).toBe(false)
    });

    test('renders correctly', () =>{
        const tree = render(<SettingsProfile/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('Font Loading Error <SettingsProfile/>', () =>{
    beforeEach(async () => {
        await AsyncStorage.setItem("dark_mode_setting", "");
        Font.useFonts.mockReturnValue([false, true])
        const [fontsLoaded, fontError] = Font.useFonts()
        expect(fontsLoaded).toBe(false)
        expect(fontError).toBe(true)
    });

    test('renders correctly', () =>{
        const tree = render(<SettingsProfile/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})