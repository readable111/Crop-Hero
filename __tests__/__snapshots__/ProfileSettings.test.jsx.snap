// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Light Mode <SettingsProfile/> renders correctly 1`] = `null`;
